frontend/src/scenes/plugins/tabs/installed/sections/UpgradeSection.tsx

```diff
@@ -0,0 +1,87 @@
+import { CaretRightOutlined, CaretDownOutlined, SyncOutlined, CloudDownloadOutlined } from '@ant-design/icons'
+import { Button, Row } from 'antd'
+import { Subtitle } from 'lib/components/PageHeader'
+import React from 'react'
+import { useActions, useValues } from 'kea'
+import { PluginSection, pluginsLogic } from 'scenes/plugins/pluginsLogic'
+import { InstalledPlugin } from '../InstalledPlugin'
+import { canInstallPlugins } from 'scenes/plugins/access'
+import { userLogic } from 'scenes/userLogic'
+
+export function UpgradeSection(): JSX.Element {
+    const { checkForUpdates, toggleSectionOpen } = useActions(pluginsLogic)
+    const { sectionsOpen } = useValues(pluginsLogic)
+    const { user } = useValues(userLogic)
+
+    const {
+        filteredPluginsNeedingUpdates,
+        pluginsNeedingUpdates,
+        checkingForUpdates,
+        installedPluginUrls,
+        updateStatus,
+        rearranging,
+        hasUpdatablePlugins,
+    } = useValues(pluginsLogic)
+
+    const upgradeButton = canInstallPlugins(user?.organization) && hasUpdatablePlugins && (
+        <Button
+            type="default"
+            icon={pluginsNeedingUpdates.length > 0 ? <SyncOutlined /> : <CloudDownloadOutlined />}
+            onClick={(e) => {
+                e.stopPropagation()
+                checkForUpdates(true)
+            }}
+            loading={checkingForUpdates}
+        >
+            {checkingForUpdates
+                ? `Checking plugin ${Object.keys(updateStatus).length + 1} out of ${
+                      Object.keys(installedPluginUrls).length
+                  }`
+                : pluginsNeedingUpdates.length > 0
+                ? 'Check again for updates'
+                : 'Check for updates'}
+        </Button>
+    )
+
+    return (
+        <>
+            <div
+                className="plugins-installed-tab-section-header"
+                onClick={() => toggleSectionOpen(PluginSection.Upgrade)}
+            >
+                <Subtitle
+                    subtitle={
+                        <>
+                            {sectionsOpen.includes(PluginSection.Upgrade) ? (
+                                <CaretDownOutlined />
+                            ) : (
+                                <CaretRightOutlined />
+                            )}
+                            {` Plugins to update (${filteredPluginsNeedingUpdates.length})`}
+                        </>
+                    }
+                    buttons={!rearranging && sectionsOpen.includes(PluginSection.Upgrade) && upgradeButton}
```

Comment:
Do we have a mechanism that checks for plugin updates without user interaction?

Reply:
We don't - could do that too

Reply:
I thought @mariusandra added something like that some time ago, but not sure

Reply:
Oh I think we check once on load, but not continuously after is what I mean

Reply:
Ah, cool, that's good. :+1: I don't think we need to check continuously. But if the "Check for updates" button is hidden when the "Plugins to update" sections is closed (which will be true most of the time – a "All your plugins are up to date. Great work!" message is not interesting), then it's going to be a seldom used, very intentional interaction. So it's good to check every now and then _without_ user interaction.

Reply:
I see yeah